js/okex.js

```diff
@@ -60,20 +70,102 @@ module.exports = class okex extends okcoinusd {
     }
 
     async fetchMarkets (params = {}) {
-        let markets = await super.fetchMarkets (params);
         // TODO: they have a new fee schedule as of Feb 7
         // the new fees are progressive and depend on 30-day traded volume
         // the following is the worst case
+
+        let futuresResponse = await this.webPostFuturesPcMarketFuturesCoin ();
+        let spotResponse = await this.webGetSpotMarketsProducts ();
+        let markets = spotResponse.data.concat (futuresResponse.data);
+        let result = [];
         for (let i = 0; i < markets.length; i++) {
-            if (markets[i]['spot']) {
-                markets[i]['maker'] = 0.0015;
-                markets[i]['taker'] = 0.0020;
+            let isFutureMarket = Object.prototype.hasOwnProperty.call (markets[i], 'contracts');
```

Comment:
this wont transpile - https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md

Reply:
@frosty00 Good catch - I amended with a `typeof` check, hope that fits the bill.